pkg/generate/config/resource.go

```diff
@@ -87,6 +90,26 @@ type ResourceConfig struct {
 	IsARNPrimaryKey bool `json:"is_arn_primary_key"`
 }
 
+// SyncedConfig instructs the code generator on how to generate functions that checks
+// whether a resource was synced or not.
+type SyncedConfig struct {
+	// When is a list of conditions that should be satisfied in order to tell whether a
+	// a resource was synced or not.
+	When []SyncedCondition `json:"when"`
+}
+
+// SyncedCondition represent one of the unique condition that should be fullfiled in
+// order to assert whether a resource is synced.
+type SyncedCondition struct {
+	// Path of the field. e.g Status.Processing
+	Path *string `json:"path"`
+	// Is contains the exact value `Path` should be equal to.
+	Is string `json:"is"`
```

Comment:
I think i will change this type to `interface{}` so we can support `bool` and `int64` types as well

Reply:
Mmmm maybe it should more have a `GoType` field. /cc @vijtrip2

Reply:
Can the string comparison not cover bool and int ?
What I mean is you can convert the value from `Path` into string and then compare with value from `Is`?

Reply:
Mmmmm yes that is possible using `json.Marshall` or `strings.Atoi/Iota`, but I think that it would be easier to just detect the Path `shape.GoType` and generate code with simple `==` tokens

Reply:
Kept the string type, and try to detect the type from the field path instead